stan/math/prim/mat/err/check_finite.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef STAN_MATH_PRIM_MAT_ERR_CHECK_FINITE_HPP
+#define STAN_MATH_PRIM_MAT_ERR_CHECK_FINITE_HPP
+
+#include <Eigen/Dense>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+
+namespace stan {
+namespace math {
+namespace {
+template <typename T, int R, int C>
+struct finite<Eigen::Matrix<T, R, C>, true> {
```

Comment:
I'd prefer the name `stan::math::internal::is_finite` to indicate the behavior of the class.